syntax = "proto3";

package org.couchers.blocking;

import "google/protobuf/empty.proto";

service Blocking {
  rpc BlockUser(BlockUserReq) returns (google.protobuf.Empty);

  rpc UnblockUser(UnblockUserReq) returns(google.protobuf.Empty);

  rpc GetBlockedUsers(google.protobuf.Empty) returns (GetBlockedUsersRes) {
    //Returns all users blocked by a given account
  }
}

message BlockUserReq {
  int64 user_id = 1;
}

message UnblockUserReq {
  int64 user_id = 1;
}

message GetBlockedUsersRes {
  repeated int64 blocked_user_ids = 1;
}
